### PR TITLE
refactor: Fixed noAssignInExpressions lint-errors

### PR DESCRIPTION
**Pull request by KenAJoh**
### Description

https://biomejs.dev/linter/rules/no-assign-in-expressions/

### CODE REVIEW

File: @navikt/core/react/src/util/i18n/get.ts [hunk 1]
```diff
@@ -38,10 +38,12 @@ export function get(
 
 function getKeypath(str: string) {
   const path: string[] = [];
-  let result: RegExpExecArray | null;
-  while ((result = OBJECT_NOTATION_MATCHER.exec(str))) {
+  let result: RegExpExecArray | null = OBJECT_NOTATION_MATCHER.exec(str);
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/util/i18n/get.ts]:
```suggestion
  let result = OBJECT_NOTATION_MATCHER.exec(str);
```

File: @navikt/core/react/src/modal/dialog-polyfill.ts [hunk 2]
```diff
@@ -632,6 +632,7 @@ dialogPolyfill.DialogManager = function () {
       var removed = [];
       records.forEach(function (rec) {
         // biome-ignore lint/suspicious/noImplicitAnyLet: Reduntant to type c in this scenario
+        // biome-ignore lint/suspicious/noAssignInExpressions: <explanation>
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/modal/dialog-polyfill.ts]: Should we just add this file to the ignore list instead?

Comment (KenAJoh) [reply @navikt/core/react/src/modal/dialog-polyfill.ts]: Sounds like a good idea 👍

Comment (KenAJoh) [reply @navikt/core/react/src/modal/dialog-polyfill.ts]: Done 🚀

**Review by HalvorHaugan — APPROVED** [review]

**Pull request merged**